HttpFs/HttpFs.fs

```diff
@@ -974,17 +974,15 @@ module Client =
     /// Uses the HttpWebRequest to get the response.
     /// HttpWebRequest throws an exception on anything but a 200-level response,
     /// so we handle such exceptions and return the response.
-    let getResponseNoException (request : HttpWebRequest) = job {
-      try
-        let! response = request.AsyncGetResponse()
-        return response :?> HttpWebResponse
-      with
-      | :? WebException as wex ->
-        if wex.Response <> null then
-          return wex.Response :?> HttpWebResponse
-        else
-          return raise wex
-    }
+    let getResponseNoException (request : HttpWebRequest) : Alt<Choice<HttpWebResponse,exn>> =
+      let inline succeed (wr : WebResponse) : Choice<HttpWebResponse,exn> = downcast wr |> Choice1Of2
+      let inline failure (ex : exn) : Choice<HttpWebResponse,exn> = Choice2Of2 ex
+      let respAlt = Alt.fromBeginEnd request.BeginGetResponse request.EndGetResponse (fun _ -> request.Abort())
+      let handleFailure (ex : exn) =
+        match ex with
+        | :? WebException as wex when wex.Response <> null -> succeed wex.Response
+        | _ -> failure ex
+      Alt.tryIn respAlt (Job.lift succeed) (Job.lift handleFailure)
```

Comment:
Is `request.EndGetResponse` the only function (called by `Alt.fromBeginEnd`) that might raise an exception?  If so, I would suggest wrapping that function to perform the conversion to `Either`, because that will likely perform better than `Alt.tryIn`.

```fsharp
(fun ar -> try request.EndGetResponse ar |> succeed with e -> failure e)
```